banzai/photometry.py

```diff
@@ -241,3 +243,39 @@ def do_stage(self, image):
         except Exception:
             logger.error(logs.format_exception(), image=image)
         return image
+
+
+class PhotometricCalibrator(Stage):
+    color_to_fit = {'gp': 'g-r', 'rp': 'r-i', 'ip': 'r-i', 'zs': 'i-z'}
+
+    def __init__(self, runtime_context):
+        super(PhotometricCalibrator, self).__init__(runtime_context)
+
+    def do_stage(self, image):
+        if image.filter not in ['gp', 'rp', 'ip', 'zs']:
+            return image
+
+        try:
+            # Get the sources in the frame
+            reference_catalog = get_reference_sources(image.meta, self.runtime_context.REFERENCE_CATALOG_URL)
+        except HTTPError as e:
+            logger.error(f'Error retrieving photometric reference catalog: {e}', image=image)
+            return image
```

Comment:
Do we want to populate a header value to indicate success/failure? Kind of like WCSERR?